File: 1. 그리디/만들 수 없는 금액.py
n = int(input())
coins = list(map(int, input().split()))
coins.sort()
# 적어도 coin을 모두 더하고 +1을 더한 금액은 만들 수 없음

target = 1

for x in coins:
  if target < x:
    break
  target += x

print(target)

File: codeEg/1. 그리디/만들 수 없는 금액.js
"use strict"

const input_data = [3, 2, 1, 1, 9]

input_data.sort((a, b) => (a - b))

let target = 1

for (let x of input_data) {
  if (target < x) break

  target += x
}

console.log(target)